src/Stateless/StateRepresentation.cs

```diff
@@ -38,45 +38,51 @@ public StateRepresentation(TState state)
 
             public bool CanHandle(TTrigger trigger)
             {
-                TriggerBehaviour unused;
+                TriggerBehaviourResult unused;
                 return TryFindHandler(trigger, out unused);
             }
 
-            public bool TryFindHandler(TTrigger trigger, out TriggerBehaviour handler)
+            public bool TryFindHandler(TTrigger trigger, out TriggerBehaviourResult handler)
```

Comment:
The tricky cases here around substates with unmet conditions, superstates with unmet conditions and so-on seem to warrant some tests - it would be really useful to illustrate these scenarios so that they're not inadvertently broken by future updates.